backend_jobs/ingestion_pipeline/unit_tests/test_resolution.py

```diff
@@ -0,0 +1,104 @@
+        'originalformat': 'jpg', 'latitude': '0', 'longitude': '0', 'height_c': '800', 'width_c': '533',
+        'url': 'https://live.staticflickr.com/882/39831840270_ba571c8254_c.jpg'}
+
+    second_image = {
+        'id': '39831840270', 'datetaken': '2018-04-22 16:41:11', 'ownername': 'Marian',
+        'originalformat': 'jpg', 'latitude': '0', 'longitude': '0', 'height_c': '800', 'width_c': '533',
+        'url': 'https://live.staticflickr.com/882/39831840270_ba571c8254.jpg'}  
+
+    flicker_provider = image_provider_flickr.FlickrProvider() 
+
+    def test_get_url_for_min_resolution_return_320_resolution(self):
+        """ Given a url with a max resolution of 800, min height of 300 and min width of 500,
+        Return a new url with closest above fit resolution of 300.
+
+        Returns:
+            Url with max resolution of 320. (n represents max resolution of 320)
+        """
+        new_url = self.flicker_provider.get_url_for_min_resolution(300, 500, self.first_image)
+        expected_url = 'https://live.staticflickr.com/882/39831840270_ba571c8254_n.jpg'
+        self.assertEqual(expected_url, new_url)
+
+    def test_get_url_for_min_resolution_return_1024_resolution(self):
+        """ Given a url with a max resolution of 800, min height of 850 and min width of 1000,
+        Return a new url with closest above fit resolution of 850.
+
+        Returns:
+            Url with max resolution of 1024. (b represents max resolution of 1024)
+        """
+        new_url = self.flicker_provider.get_url_for_min_resolution(850, 1000, self.first_image)
+        expected_url = 'https://live.staticflickr.com/882/39831840270_ba571c8254_b.jpg'
+        self.assertEqual(expected_url, new_url)
+
+    def test_get_url_for_min_resolution_return_240_resolution(self):
+        """ Given a url with a max resolution of 500, min height of 500 and min width of 200,
+        Return a new url with closest above fit resolution of 200.
+
+        Returns:
+            Url with max resolution of 240. (m represents max resolution of 240)
+        """
+        flicker_provider = image_provider_flickr.FlickrProvider()
+        new_url = flicker_provider.get_url_for_min_resolution(500, 200, self.second_image)
+        expected_url = 'https://live.staticflickr.com/882/39831840270_ba571c8254_m.jpg'
+        self.assertEqual(expected_url, new_url)
+
+    def test_get_url_for_min_resolution_return_800_resolution(self):
+        """ Given a url with a max resolution of 500, min height of 1000 and min width of 750,
+        Return a new url with closest above fit resolution of 750.
+
+        Returns:
+            Url with max resolution of 800. (c represents max resolution of 800)
+        """
+        flicker_provider = image_provider_flickr.FlickrProvider()
+        new_url = flicker_provider.get_url_for_min_resolution(1000, 750, self.second_image)
+        expected_url = 'https://live.staticflickr.com/882/39831840270_ba571c8254_c.jpg'
```

Comment:
why do we get back an image with longest edge = 800 ? (c is for 800)
The longest edge should be 1000, right?
(here and other places)